src/main/java/com/nha/abdm/wrapper/hiu/hrp/dataTransfer/HIUFacadeHealthInformationService.java

```diff
@@ -79,10 +89,28 @@ public FacadeResponse healthInformation(
       throws InvalidAlgorithmParameterException, NoSuchAlgorithmException, NoSuchProviderException,
           IllegalDataStateException {
     if (Objects.isNull(hiuClientHealthInformationRequest)) {
-      return FacadeResponse.builder().httpStatusCode(HttpStatus.BAD_REQUEST).build();
+      return FacadeResponse.builder()
+          .clientRequestId(hiuClientHealthInformationRequest.getRequestId())
+          .httpStatusCode(HttpStatus.BAD_REQUEST)
+          .build();
+    }
+    String consentId = hiuClientHealthInformationRequest.getConsentId();
+    // Fetching dateRange from consent present in db.
+    Consent consentDetails =
+        patientService.getConsentDetails(
+            consentPatientService.findMappingByConsentId(consentId).getAbhaAddress(), consentId);
+    if (Objects.isNull(consentDetails)) {
+      return FacadeResponse.builder()
+          .clientRequestId(hiuClientHealthInformationRequest.getRequestId())
+          .error(
+              ErrorResponse.builder()
+                  .code(400)
+                  .message("ConsentId not found in consentPatient")
```

Comment:
lets give the message as "Consent not found in database."

Reply:
Modified to database.